xapian-core/api/positioniterator.cc

```diff
@@ -76,6 +78,20 @@ PositionIterator::operator=(const PositionIterator & o)
     RETURN(*this);
 }
 
+PositionIterator::PositionIterator(PositionIterator && o)
+    : internal(o.internal)
+{
+    o.internal = nullptr;
+}
```

Comment:
This should be OK to have inline in the public API header (`include/xapian/positioniterator.h`).

That's beneficial for a trivial implementation like this one as it means that the compiler can inline it and optimise the result.  It also means we don't create another external symbol in the shared library (the fewer external symbols there are, the quicker it loads).  One extra isn't a big deal by itself, but where they can be easily avoided we try to.

Reply:
Thanks for the tip. Done.